dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -407,6 +408,90 @@ def new_integration
     render template: Policies::Lti.early_access? ? 'lti/v1/integrations/early_access' : 'lti/v1/integrations'
   end
 
+  # GET /lti/v1/dynamic_registration
+  # Initiate LTI Dynamic Registration flow
+  def new_registration
+    openid_configuration_url = params[:openid_configuration]
+    registration_token = params[:registration_token]
+    begin
+      openid_config = JSON.parse(HTTParty.get(openid_configuration_url).body).symbolize_keys
+    rescue => exception
+      Honeybadger.notify(exception, context: {message: 'Error getting LMS openid_configuration'})
+      return render status: :internal_server_error
+    end
+
+    registration_id = SecureRandom.uuid
+    puts registration_id
+    registration_data = {
+      registration_token: registration_token,
+      registration_endpoint: openid_config[:registration_endpoint],
+      issuer: openid_config[:issuer],
+      registration_id: registration_id,
+    }
+
+    write_cache(registration_id, registration_data, 1.hour)
+    # render form view with any data we'd like to offer as configurable
+    # TEMP placeholder HTML
+    render html: <<-HTML.html_safe
+      <!DOCTYPE html>
+      <html>
+      <head>
+        <title>Simple HTML Page</title>
+      </head>
+      <body>
+        <h1>Hello, World!</h1>
+      </body>
+      </html>
+    HTML
+  end
+
+  # POST /lti/v1/dynamic_registration
+  # Initiate LTI Dynamic Registration Request and create LTI Integration
+  def create_registration
+    admin_email = params[:email]
+    # Get cached values
+    cache_key = params[:registration_id]
+    registration_data = read_cache(cache_key)
+
+    platform = Policies::Lti.find_platform_by_issuer(registration_data[:issuer])
+    # Create registration request
+    body = Policies::Lti::DYNAMIC_REGISTRATION_CONFIG
+    headers = {
+      'Content-Type' => 'application/json',
+      'Authorization' => "Bearer #{registration_data[:registration_token]}"
+    }
+
+    registration = nil
+    begin
+      response = HTTParty.post(registration_data[:registration_endpoint], body: body.to_json, headers: headers)
+      registration = JSON.parse(response.body).symbolize_keys
+    rescue => exception
+      Honeybadger.notify(exception, context: {message: 'Error creating dynamic registration'})
+      return render status: :internal_server_error
+    end
+
+    existing_integration = Queries::Lti.get_lti_integration(platform[:issuer], registration[:client_id])
+    if existing_integration.nil?
+      Services::Lti.create_lti_integration(
+        name: 'Test Dynamic Registration',
```

Comment:
TODO: Change name

Reply:
UPDATE: Ali wants to use the value from `https://canvas.instructure.com/lti/account_name`. Add this to the object we are caching.